### PR TITLE
Below Zero Saving Cache

### PR DESCRIPTION
**Pull request by PrimeSonic**
- Adds a new patch that allows methods to be invoked when the game is saving
- This will be used by the EnumCacheManager so that the file is only saved when the player manually saves

### CODE REVIEW

File: SMLHelper/Utility/SaveUtils.cs [hunk 1]
```diff
@@ -12,5 +15,23 @@ public static string GetCurrentSaveDataDir()
...
+        public static void RegisterOnSaveEvent(Action onSaveAction)
+        {
+            IngameMenuPatcher.OnSaveEvents += onSaveAction;
+        }
+
+        /// <summary>
+        /// Removes a method that would be invoked whenever the game is saved via the in-game menu.
+        /// </summary>
+        /// <param name="onSaveAction">The method invoked.</param>
+        public static void DeregisterOnSaveEvent(Action onSaveAction)
```
Comment (AlexejheroYTB) [inline SMLHelper/Utility/SaveUtils.cs]: Unregister

File: SMLHelper/Utility/SaveUtils.cs [hunk 2]
```diff
@@ -12,5 +15,24 @@ public static string GetCurrentSaveDataDir()
         {
             return SaveLoadManager.temporarySavePath;
         }
+
+        /// <summary>
+        /// Registers a simple <see cref="Action"/> method to invoke whenever the player saves the game via the in game menu.
+        /// </summary>
+        /// <param name="onSaveAction">The method to invoke.</param>
+        public static void RegisterOnSaveEvent(Action onSaveAction)
```
Comment (AlexejheroYTB) [inline SMLHelper/Utility/SaveUtils.cs]: This should be moved to IngameMenuHandler

Comment (PrimeSonic) [reply SMLHelper/Utility/SaveUtils.cs]: Created `IngameMenuHandler` and moved the new methods to this space.

File: SMLHelper/Handlers/IngameMenuHandler.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        /// <param name="onSaveAction">The method invoked.</param>
+        public static void UnregisterOnSaveEvent(Action onSaveAction)
+        {
+            Main.UnregisterOnSaveEvent(onSaveAction);
+        }
+
+        /// <summary>
+        /// Registers a simple <see cref="Action"/> method to invoke whenever the player saves the game via the in game menu.
+        /// </summary>
+        /// <param name="onSaveAction">The method to invoke.</param>
```
Comment (AlexejheroYTB) [inline SMLHelper/Handlers/IngameMenuHandler.cs]: The summary needs to reflect the fact that this method is one-time. Also, I think you could merge `RegisterOnSaveEvent` with `RegisterOneTimeUseOnSaveEvent`, and have an optional `bool` parameter that defaults to `false` dictating whether that event is one-time or not.

Comment (PrimeSonic) [reply SMLHelper/Handlers/IngameMenuHandler.cs]: I considered merging the two methods with an optional parameter, but then it complicated the documentation and explanation of the `UnregisterOnSaveEvent` method.

To keep things simple, I thought it best to just make a separate method with clear documentation.

**Pull request merged**